backend/src/controllers/groups-controller.ts

```diff
@@ -77,6 +84,21 @@ export const updateName = async (req: Request, res: Response) =>{
     if(existingGroup){
         return res.status(409).json({ error: "This group name is already taken"});
     }
+    //Checks if UserId exists
+    if(!req.token?.userId){
+        return res.status(422).json({error: "UserId does not exist"});
+    }
+
+    const groupOwner = req.body.userId;
+    const findGroupOwnerById = Group.find({ userId: groupOwner });
+    //Makes sure user is owner of the item
+    if (!findGroupOwnerById){
```

Comment:
Related: this condition will always evaluate to true when using `find`, since `if (anyList)` always will evaluate to true.

Use the above fix to fix this issue.